### PR TITLE
{vis} [intel/2016a] JWM 2.3.5

### PR DESCRIPTION
**Pull request by wpoely86**

### CODE REVIEW

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 5 out of 6 (6 easyconfigs in this PR)
node2703.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/50dd2d6b7fb4e68681a3aeb3c038ac95 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 5 out of 6 (6 easyconfigs in this PR)
node2703.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/b3b8066ff059ff72f75219d03f254c56 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
node2703.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/4f4ad98d02fc6c711eeb35ef39999a19 for a full test report.

**Comment by wpoely86** [comment]
@boegel please review

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 12 out of 12 (6 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/440a1be5e184df08c401896b158fe57d for a full test report.

File: easybuild/easyconfigs/j/JWM/JWM-2.3.5-intel-2016a.eb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+easyblock = 'ConfigureMake'
+
+name = 'JWM'
+version = '2.3.5'
+
+homepage = 'https://joewing.net/projects/jwm/'
+description = """JWM is a light-weight window manager for the X11 Window System."""
```
Comment (fgeorgatos) [inline easybuild/easyconfigs/j/JWM/JWM-2.3.5-intel-2016a.eb]: window manager? WINDOW MANAGER?! aarggh... I knew this day would come one day :)

first visual review ok, thanks for the contribution!

Comment (wpoely86) [inline easybuild/easyconfigs/j/JWM/JWM-2.3.5-intel-2016a.eb]: It's a historic step forward!

I'm working on the full VNC server so we will soon have Xorg too :wink:

File: easybuild/easyconfigs/f/FLTK/FLTK-1.3.3-intel-2016a.eb [hunk 2]
```diff
@@ -17,6 +17,7 @@ toolchainopts = {'optarch': True, 'pic': True}
 sources = ['%(namelower)s-%(version)s-source.tar.gz']
 source_urls = ['http://fltk.org/pub/%(namelower)s/%(version)s/']
 
+configopts = '--enable-shared --enable-threads --enable-xft'
```
Comment (boegel) [inline easybuild/easyconfigs/f/FLTK/FLTK-1.3.3-intel-2016a.eb]: @wpoely86 please do this in all FLTK easyconfig (separate PR would be best)?

File: easybuild/easyconfigs/l/librsvg/librsvg-2.40.15-intel-2016a.eb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+dependencies = [
+    ('Gdk-Pixbuf', '2.35.1'),
+    ('libcroco', '0.6.11'),
+    ('Pango', '1.40.1'),
+    ('cairo', '1.14.6', '-GLib-2.48.0'),
+]
+
+sanity_check_paths = {
+    'files': ['bin/rsvg-convert', 'lib/librsvg-2.%s' % SHLIB_EXT, 'lib/librsvg-2.a'],
```
Comment (boegel) [inline easybuild/easyconfigs/l/librsvg/librsvg-2.40.15-intel-2016a.eb]: use `%(version_major)s` rather than hardcoding `2`?

File: easybuild/easyconfigs/x/XZ/XZ-5.2.2-intel-2016a.eb [hunk 4]
```diff
@@ -13,9 +13,6 @@ source_urls = ['http://tukaani.org/xz/']
 
 builddependencies = [
     ('Autotools', '20150215'),
-]
-
-dependencies = [
```
Comment (boegel) [inline easybuild/easyconfigs/x/XZ/XZ-5.2.2-intel-2016a.eb]: @wpoely86 sure about this?

if so, do this in a separate PR for all XZ easyconfigs (incl. this one)

Comment (wpoely86) [inline easybuild/easyconfigs/x/XZ/XZ-5.2.2-intel-2016a.eb]: `ldd` show no link to gettext and the `pkg-config --cflags --libs` neither.

Comment (boegel) [inline easybuild/easyconfigs/x/XZ/XZ-5.2.2-intel-2016a.eb]: and XZ doesn't call out to any of the binaries provided by `gettext`?

```
autopoint  gettext     gettext.sh  msgcat  msgcomm  msgen    msgfilter	msggrep  msgmerge  msguniq   recode-sr-latin
envsubst   gettextize  msgattrib   msgcmp  msgconv  msgexec  msgfmt	msginit  msgunfmt  ngettext  xgettext
```

Comment (wpoely86) [inline easybuild/easyconfigs/x/XZ/XZ-5.2.2-intel-2016a.eb]:
```
checking for GNU gettext in libc... yes
checking whether to use NLS... yes
checking where the gettext function comes from... libc
```
I'm guessing it takes it from glibc?

Comment (boegel) [inline easybuild/easyconfigs/x/XZ/XZ-5.2.2-intel-2016a.eb]: If it's there, sure, but what if it's not there (older glibc)?

I wouldn't change this in this PR, and certainly not for just a single XZ easyconfig.

Comment (wpoely86) [inline easybuild/easyconfigs/x/XZ/XZ-5.2.2-intel-2016a.eb]: then this PR cannot be merged until 2016b. It's one big versionsuffix conflict without this.

File: easybuild/easyconfigs/l/libcroco/libcroco-0.6.11-intel-2016a.eb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+toolchain = {'name': 'intel', 'version': '2016a'}
+
+sources = [SOURCE_TAR_XZ]
+source_urls = ['http://ftp.gnome.org/pub/GNOME/sources/libcroco/%(version_major_minor)s/']
+
+dependencies = [
+    ('zlib', '1.2.8'),
+    ('libxml2', '2.9.3'),
+    ('GLib', '2.48.0'),
+    ('XZ', '5.2.2',)
```
Comment (boegel) [inline easybuild/easyconfigs/l/libcroco/libcroco-0.6.11-intel-2016a.eb]: if you added this because the source file is `.xz`, it's kind of futile, cfr. https://github.com/hpcugent/easybuild-framework/issues/1376

also, in that case, it should be a build dep

File: easybuild/easyconfigs/l/librsvg/librsvg-2.40.15-intel-2016a.eb [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+homepage = 'https://wiki.gnome.org/action/show/Projects/LibRsvg'
+description = """librsvg is a library to render SVG files using cairo."""
+
+toolchain = {'name': 'intel', 'version': '2016a'}
+
+sources = [SOURCE_TAR_XZ]
+source_urls = ['https://download.gnome.org/sources/librsvg/%(version_major_minor)s']
+
+# this loader wants to install in the directory of Gdk-Pixbuf itself, so disable
+configopts = '--disable-pixbuf-loader'
```
Comment (boegel) [inline easybuild/easyconfigs/l/librsvg/librsvg-2.40.15-intel-2016a.eb]: move down pls

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 5 out of 6 (6 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/4a71e7ff30614be05d2a413662a31a7b for a full test report.

File: easybuild/easyconfigs/f/FLTK/FLTK-1.3.3-intel-2016a.eb [hunk 7]
```diff
@@ -27,10 +28,11 @@ dependencies = [
     ('libpng', '1.6.21'),
     ('libjpeg-turbo', '1.4.2'),
     ('xprop', '1.2.2'),
+    ('libXft', '2.3.2', '-fontconfig-2.11.95'),
```
Comment (boegel) [inline easybuild/easyconfigs/f/FLTK/FLTK-1.3.3-intel-2016a.eb]: @wdpypere seems like this introduced a conflict?

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 42 out of 43 (6 easyconfigs in this PR)
node2648.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/af283e8dff043816b8c21c911ece03dd for a full test report.

File: easybuild/easyconfigs/f/FLTK/FLTK-1.3.3-intel-2016a.eb (hunk 7 above)
Comment (wdpypere) [inline easybuild/easyconfigs/f/FLTK/FLTK-1.3.3-intel-2016a.eb]: que?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/548ec12e5ca0b110adbe538168b24a43 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 43 out of 43 (6 easyconfigs in this PR)
node2648.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/81a21080616d857572f7a0a8115a787d for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 4 out of 6 (6 easyconfigs in this PR)
node2428.golett.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/763a768473b59176273078cd3f4dc362 for a full test report.

**Comment by wpoely86** [comment]
@boegel that's very weird. It works on my end but not yours?

**Comment by boegel** [comment]
@wpoely86 some of my modules are screwed up, they were installed against the wrong Python, it's not a problem with this PR... Fixing it, will submit a new test report, but PR should be good to go.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
node2428.golett.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/c385e8647b3e69eba0abc8ac670c2f9f for a full test report.

**Comment by boegel** [comment]
lgtm

**Comment by boegel** [comment]
Going in, thanks @wpoely86!
